giraffe/src/utils/getTicks.ts

```diff
@@ -112,20 +114,93 @@ export const getTicks = (
     : getOptimalTicks(domain, rangeLength, tickSize)
 }
 
-const getMemoizedVerticalTicks = memoizeOne(
-  (
-    domain: number[],
-    rangeLength: number,
-    tickSize: number,
-    columnKey: string
-  ) => getTicks(domain, rangeLength, tickSize, columnKey)
-)
+/*
+  generateTicks gives control to the user over placement, number, and interval of ticks
+    defers to the user's judgement on spacing (tick labels may overlap)
+*/
+export const generateTicks = (
+  domain: number[],
+  columnKey: string,
+  totalTicks: number,
+  tickStart: number,
+  tickStep: number
+): number[] => {
+  const generatedTicks = []
+  const [start, end] = domain
+  const stepStart = isFiniteNumber(tickStart) ? tickStart : start
+  const step = isFiniteNumber(tickStep)
+    ? tickStep
+    : (end - stepStart) / (totalTicks + 1)
+  const tickCountLimit = isFiniteNumber(totalTicks) ? totalTicks : Infinity
+
+  let counter = isFiniteNumber(tickStart) ? 0 : 1
+  let generatedTick = stepStart + step * counter
+
+  if (
+    tickStep !== 0 &&
+    (isFiniteNumber(totalTicks) || isFiniteNumber(tickStep))
+  ) {
+    while (
+      generatedTick >= start &&
+      generatedTick <= end &&
+      generatedTicks.length < tickCountLimit
+    ) {
+      if (columnKey === TIME) {
+        generatedTicks.push(new Date(generatedTick))
+      } else {
+        generatedTicks.push(generatedTick)
+      }
+      counter += 1
+      generatedTick = stepStart + step * counter
+    }
+  }
+  return generatedTicks
+}
+
+/*
+  ticks can be either
+    - generated: user's defined parameters and judgmenet for spacing
+    - calculated: Giraffe's determination and spacing requirements
+*/
+const getTicks = (
+  domain: number[],
+  rangeLength: number,
+  tickSize: number,
+  columnKey: string,
+  totalTicks?: number,
+  tickStart?: number,
+  tickStep?: number
+) => {
+  const [start = 0, end = 0] = domain
+  if (isFiniteNumber(totalTicks) || isFiniteNumber(tickStep)) {
```

Comment:
should this `||` be an `&&`?

Reply:
No. It is correct as `||`. I will demo this to show off the flexibility.

Reply:
Sure, will do.